src/sagemaker/jumpstart/utils.py

```diff
@@ -1127,43 +1129,91 @@ def get_metrics_from_deployment_configs(
-
-            data["Config Name"].append(deployment_config.deployment_config_name)
-            instance_type_to_display = (
-                f"{current_instance_type} (Default)"
-                if index == 0
-                and current_instance_type == deployment_config.deployment_args.default_instance_type
-                else current_instance_type
+        for current_instance_type, current_instance_type_metrics in benchmark_metrics.items():
+            instance_type_rate, concurrent_users = _normalize_benchmark_metrics(
+                current_instance_type_metrics
             )
-            data["Instance Type"].append(instance_type_to_display)
-
-            for metric in current_instance_type_metrics:
-                column_name = f"{metric.name} ({metric.unit})"
-
-                if metric.name.lower() == "instance rate":
-                    if column_name not in instance_rate_data:
-                        instance_rate_data[column_name] = []
-                    instance_rate_data[column_name].append(metric.value)
-                else:
-                    if column_name not in data:
-                        data[column_name] = []
-                    for _ in range(len(data[column_name]), inner_index):
-                        data[column_name].append(" - ")
+
+            for concurrent_user, metrics in concurrent_users.items():
+                instance_type_to_display = (
+                    f"{current_instance_type} (Default)"
+                    if index == 0
+                    and int(concurrent_user) == 1
+                    and current_instance_type
+                    == deployment_config.deployment_args.default_instance_type
+                    else current_instance_type
+                )
+
+                instance_rate_column_name = f"{instance_type_rate.name} ({instance_type_rate.unit})"
+                instance_rate_data[instance_rate_column_name] = instance_rate_data.get(
+                    instance_rate_column_name, []
+                )
+
+                data["Config Name"].append(deployment_config.deployment_config_name)
+                data["Instance Type"].append(instance_type_to_display)
+                data["Concurrent Users"].append(concurrent_user)
+                instance_rate_data[instance_rate_column_name].append(instance_type_rate.value)
+
+                for metric in metrics:
+                    column_name = _normalize_benchmark_metric_column_name(metric.name)
+                    data[column_name] = data.get(column_name, [])
                     data[column_name].append(metric.value)
 
     data = {**data, **instance_rate_data}
     return data
 
 
+def _normalize_benchmark_metric_column_name(name: str) -> str:
+    """Normalizes benchmark metric column name.
+
+    Args:
+        name (str): Name of the metric.
+    Returns:
+        str: Normalized metric column name.
+    """
+    if "latency" in name.lower():
+        name = "Latency for each user (TTFT in ms)"
```

Comment:
could we use the metric unit from metadata directly?

Reply:
We can, only if it's updated to the desired. Here is what's now
```
{
                        "name": "latency",
                        "value": "36",
                        "unit": "ms/token",
                        "concurrency": "2"
},
```